test/spec/modules/exadsBidAdapter_spec.js

```diff
@@ -0,0 +1,498 @@
+import { expect } from 'chai';
+import { spec, imps } from 'modules/exadsBidAdapter.js';
+import { BANNER, NATIVE, VIDEO } from '../../../src/mediaTypes.js';
+
+describe('exadsBidAdapterTest', function () {
+  const bidder = 'exadsadserver';
+
+  const partners = {
+    RTB_2_4: 'rtb_2_4'
+  }
+
+  const imageBanner = {
+    mediaTypes: {
+      banner: {
+        sizes: [300, 250]
+      }
+    },
+    bidder: bidder,
+    params: {
+      zoneId: 5147485,
+      fid: '829a896f011475d505a0d89cfdd1af8d9cdb07ff',
+      partner: partners.RTB_2_4,
+      siteId: '12345',
+      catIab: ['IAB25-3'],
+      userIp: '0.0.0.0',
+      userId: '',
+      country: 'IRL',
+      impressionId: '123456',
+      keywords: 'lifestyle, humour',
+      bidfloor: 0.00000011,
+      bidfloorcur: 'EUR',
+      mimes: ['image/jpg'],
+      image_output: 'html',
+      endpoint: 'test.com'
+    }
+  };
+
+  const native = {
+    mediaTypes: {
+      native: {
+        ortb: {
+          assets: [{
+            id: 3,
+            required: 1,
+            title: {
+              len: 124
+            }
+          },
+          {
+            id: 2,
+            data: {
+              type: 1,
+              len: 50
+            }
+          },
+          {
+            id: 1,
+            required: 1,
+            img: {
+              type: 3,
+              w: 300,
+              h: 300,
+            }
+          }]
+        }
+      },
+    },
+    bidder: bidder,
+    params: {
+      zoneId: 5147485,
+      fid: '829a896f011475d505a0d89cfdd1af8d9cdb07ff',
+      partner: partners.RTB_2_4,
+      siteId: '12345',
+      catIab: ['IAB25-3'],
+      userIp: '0.0.0.0',
+      userId: '',
+      country: 'IRL',
+      impressionId: '123456',
+      keywords: 'lifestyle, humour',
+      bidfloor: 0.00000011,
+      bidfloorcur: 'EUR',
+      native: {
+        plcmtcnt: 4,
+      },
+      endpoint: 'test.com'
+    }
+  };
+
+  const instream = {
+    mediaTypes: {
+      video: {
+        mimes: ['video/mp4'],
+        protocols: [3, 6],
+      }
+    },
+    bidder: bidder,
+    params: {
+      zoneId: 5147485,
+      fid: '829a896f011475d505a0d89cfdd1af8d9cdb07ff',
+      partner: partners.RTB_2_4,
+      siteId: '12345',
+      catIab: ['IAB25-3'],
+      userIp: '0.0.0.0',
+      userId: '',
+      country: 'IRL',
+      impressionId: '123456',
+      keywords: 'lifestyle, humour',
+      bidfloor: 0.00000011,
+      bidfloorcur: 'EUR',
+      stream: {
+        video: {
+          mimes: ['video/mp4']
+        },
+        protocols: [
+          3,
+          6
+        ],
+        ext: {
+          video_cta: 0
+        }
+      },
+      endpoint: 'test.com',
+    }
+  };
+
+  describe('bidRequestValidity', function () {
```

Comment:
The "describe" statements should describe what the single unit tests of this group are testing.
All "it" statements in test files should describe exactly what is expected of that specific test with a structure similar to it('should + verb (make/do/call/test...) ...', ...). I would suggest adapting the existing tests for better readability and understanding of what is expected in each.

Reply:
Adapted all it descriptions.